iptables/src/iptables_policies.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/algorithm/string/trim.hpp>
+#include <osquery/sdk.h>
+
+#include <trailofbits/extutils.h>
+
+#include "iptables_policies.h"
+#include "utils.h"
+
+using namespace osquery;
+
+namespace trailofbits {
+osquery::QueryData IptablesPoliciesTable::generate(
+    osquery::QueryContext& context) {
+  osquery::QueryData results;
+
+  for (auto& table : getIptablesNames()) {
+    genIptablesPolicy(table, results);
+  }
+
+  return results;
+}
+
+void IptablesPoliciesTable::genIptablesPolicy(const std::string& filter,
+                                              osquery::QueryData& results) {
+  // Initialize the access to iptc
+  auto handle = iptc_init(filter.c_str());
+  if (handle == nullptr) {
+    return;
+  }
+
+  // Iterate through chains
+  for (auto chain = iptc_first_chain(handle); chain != nullptr;
+       chain = iptc_next_chain(handle)) {
+    // NOTE(ww): Only built-in chains have default policies,
+    // and so iptc_get_policy only works on them.
+    if (!iptc_builtin(chain, handle)) {
+      TLOG << "Skipping non-built-in chain: " << chain;
+      continue;
+    }
+
+    Row r;
+
+    r["filter_name"] = TEXT(filter);
+    r["chain"] = TEXT(chain);
+
+    ipt_counters counters;
+    auto policy = iptc_get_policy(chain, &counters, handle);
```

Comment:
can this call fail? same for ip6 version

Reply:
Seems to be unspecified. It shouldn't ever fail (unless you give it a non-built-in chain, which we guard for), but it doesn't hurt to be careful. I'll add a check.